src/main/kotlin/nl/info/zac/documentcreation/BpmnDocumentCreationService.kt

```diff
@@ -0,0 +1,133 @@
+import net.atos.zac.util.MediaTypes
+import nl.info.client.smartdocuments.model.document.OutputFormat
+import nl.info.client.smartdocuments.model.document.Selection
+import nl.info.client.smartdocuments.model.document.SmartDocument
+import nl.info.client.smartdocuments.model.document.Variables
+import nl.info.client.zgw.util.extractUuid
+import nl.info.zac.authentication.LoggedInUser
+import nl.info.zac.documentcreation.converter.DocumentCreationDataConverter
+import nl.info.zac.documentcreation.model.BpmnDocumentCreationDataAttended
+import nl.info.zac.documentcreation.model.DocumentCreationAttendedResponse
+import nl.info.zac.identity.model.getFullName
+import nl.info.zac.smartdocuments.SmartDocumentsService
+import nl.info.zac.smartdocuments.SmartDocumentsTemplatesService
+import nl.info.zac.util.AllOpen
+import nl.info.zac.util.NoArgConstructor
+import java.time.ZonedDateTime
+import java.util.UUID
+import java.util.logging.Level
+import java.util.logging.Logger
+
+/***
+ * Document creation service for BPMN zaken
+ *
+ * BPMN forms (form.io) maintain inside them the SmartDocuments group name and a mapping
+ * template -> informatieobjecttype. A selector in the form provides the SmartDocuments template name.
+ *
+ * This service directly calls SmartDocuments with the provided template and template group names (without consulting
+ * the CMMN-only mapping) and uses the informatieobjecttype UUID provided by the BPMN form to call OpenZaak.
+ */
+@NoArgConstructor
+@ApplicationScoped
+@AllOpen
+@Suppress("LongParameterList")
+class BpmnDocumentCreationService @Inject constructor(
+    private val smartDocumentsService: SmartDocumentsService,
+    private val smartDocumentsTemplatesService: SmartDocumentsTemplatesService,
+    private val documentCreationService: DocumentCreationService,
+    private val documentCreationDataConverter: DocumentCreationDataConverter,
+    private val loggedInUserInstance: Instance<LoggedInUser>,
+) {
+    companion object {
+        private const val SMART_DOCUMENTS_REDIRECT_URL_BASE =
+            "rest/document-creation/smartdocuments/bpmn-callback/zaak/{zaakUuid}"
+
+        private val LOG = Logger.getLogger(BpmnDocumentCreationService::class.java.name)
+    }
+
+    fun createBpmnDocumentAttended(
+        documentCreationDataAttended: BpmnDocumentCreationDataAttended
+    ): DocumentCreationAttendedResponse =
+        documentCreationDataConverter.createData(
+            loggedInUser = loggedInUserInstance.get(),
+            zaak = documentCreationDataAttended.zaak,
+            taskId = documentCreationDataAttended.taskId
+        ).runCatching {
+            createSmartDocumentForAttendedFlow(documentCreationDataAttended).let {
+                smartDocumentsService.createDocumentAttended(
+                    data = this,
+                    smartDocument = it
+                )
+            }
+        }.onFailure {
+            LOG.log(Level.WARNING, "Failed to create document with attended flow", it)
```

Comment:
```suggestion
            LOG.log(Level.WARNING, "Failed to create SmartDocuments document for zaak with uuid: '${documentCreationDataAttended.zaak.uuid}' using attended flow", it)
```

? Important to log enough information to be able to do effective troubleshooting. The zaak uuid is always helpful if we have it.